### PR TITLE
Add QuickSortDualPivot.java

### PR DESCRIPTION
**Pull request by the-cybersapien**
Added Java implementation for Quick Sort Algorithm using Two Pivots.

### CODE REVIEW

File: Sorting Algorithms/Quick Sort Algorithm/Java/QuickSortDualPivot.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+===== ===== ===== */
+
+import java.util.Scanner;
+
+public class QuickSortDualPivot
+{
+	
+	public static void main(String[] args) {
+
+		// Initialize the Scanner Object to take input
```
Comment (VamsiSangam) [inline Sorting Algorithms/Quick Sort Algorithm/Java/QuickSortDualPivot.java]: Please correct alignment here. There's no need for a new line in line 18.

File: Sorting Algorithms/Quick Sort Algorithm/Java/QuickSortDualPivot.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    	ht++;
+    	int temp = array[ht];
+    	array[ht] = array[high];
+    	array[high] = temp;
+    	lt--;
+    	temp = array[lt];
+    	array[lt] = array[low];
+    	array[low] = temp;
+
+    	//Recursively do the same to the sub-arrays
```
Comment (VamsiSangam) [inline Sorting Algorithms/Quick Sort Algorithm/Java/QuickSortDualPivot.java]: Please correct spacing

File: Sorting Algorithms/Quick Sort Algorithm/Java/QuickSortDualPivot.java (hunk 1 above)
Comment (the-cybersapien) [reply Sorting Algorithms/Quick Sort Algorithm/Java/QuickSortDualPivot.java]: Done.

**Comment by VamsiSangam** [comment]
Please correct the alignment of comment in [line 18](https://github.com/the-cybersapien/theoryofprogramming/blob/15f1b5a72d0b4cbc924eadd8f55c925e5a7e26a5/Sorting%20Algorithms/Quick%20Sort%20Algorithm/Java/QuickSortDualPivot.java#L18). It starts with extra spacing.

**Comment by the-cybersapien** [comment]
I'm not sure why this is, in my text editor, it's indented properly, on github, it is being messed up.
I'll realign it

**Comment by VamsiSangam** [comment]
Give it a try. If still nothing happens, then let's ignore it. Its a minor issue anyway 😄

**Comment by VamsiSangam** [comment]
It happens to me too. But usually even if it is messed up in GitHub, when it is put up on the website it shows the code with proper alignment.

**Comment by the-cybersapien** [comment]
Okay, fixed it by using github edit itself.
Even removing and then reindenting the whole file showed no change in the git on local system.

**Review by VamsiSangam — APPROVED** [review]

**Comment by VamsiSangam** [comment]
Cool. Merging PR now. You should be able to see your code live on the [quick sort](http://theoryofprogramming.com/2015/01/20/quick-sort-algorithm/) post in a few minutes.

**Pull request merged**